lms/djangoapps/discussion_api/api.py

```diff
@@ -72,6 +74,19 @@ def _get_thread_and_context(request, thread_id, parent_id=None, retrieve_kwargs=
         raise Http404
 
 
+def _user_is_author_or_privileged(request, cc_thread, context):
+    """
+    Check if the user is the author of a thread or a privileged user. Such users
+    may update or delete said thread.
+
+    Raises:
+        Raises PermissionDenied exception if user is not author or privleged.
+    """
+    is_author = str(request.user.id) == cc_thread["user_id"]
```

Comment:
You can use context["cc_requester"]["id"] instead of request.user.id, which would make the request parameter unnecessary